### PR TITLE
Fail build on compiler warnings

### PR DESCRIPTION
**Pull request by serhii-lekariev**
This PR forces the build to fail on compiler warnings. This includes Error Prone warnings.

Before, warnings were logged, but the build kept running.

### CODE REVIEW

**Comment by serhii-lekariev** [comment]
@armiol, PTAL.

**Comment by serhii-lekariev** [comment]
> As per discussion, let's have this PR merged once we have a PR to `core-java` with the issues addressed.

I've reverted the Error Prone version bump. I suggest that this branch is merged in its current state, and the version is bumped in a follow-up PR, together with the fixes in the existing repositories.

**Comment by serhii-lekariev** [comment]
Let's actually not do that yet, and discuss the warning in `base` first.

File: gradle/javac-args.gradle [hunk 1]
```diff
@@ -33,8 +33,11 @@ tasks.withType(JavaCompile) {
 
     // Explicitly states the encoding of the source and test source files, ensuring
     // correct execution of the `javac` task.
+    //
+    // Also promote compiler warnings to errors, so that build fails on warnings.
```
Comment (armiol) [inline gradle/javac-args.gradle:37]: `promotes`.

**Pull request merged**